app/server/lib/OIDCConfig.ts

```diff
@@ -159,9 +164,16 @@ export class OIDCConfig {
       const email = userInfo.email;
       const fname = userInfo.given_name ?? '';
       const lname = userInfo.family_name ?? '';
+
+      const fullname = `${fname} ${lname}`.trim() ||
+        // display_name is provided by Authelia for example.
+        userInfo.display_name ||
+        // name is provided by Gitlab for example.
+        userInfo.name;
```

Comment:
Would it be important to fall back to a string (and not, say, `undefined`)? If none of these options are available, one more reasonable fallback would be `email.split("@", 1)[0]`.